lib/js-yaml/dumper.js

```diff
@@ -301,7 +307,7 @@ function chooseScalarStyle(string, singleLineOnly, indentPerLevel, lineWidth, te
       ? STYLE_PLAIN : STYLE_SINGLE;
   }
   // Edge case: block indentation indicator can only have one digit.
-  if (string[0] === ' ' && indentPerLevel > 9) {
+  if (needIndentIndicator(string) && indentPerLevel > 9) {
```

Comment:
Looking back, an optimization would be to short-circuit by first comparing `indentPerLevel > 9`, because it's almost always something like 2, 4, or 8. Likely not a big deal though.

Reply:
Makes sense. I'm happy to change it, or you can.